src/p2p/mod.rs

```diff
@@ -1,27 +1,84 @@
+use std::collections::HashMap;
 use std::str::FromStr;
+use std::sync::Arc;
 
+use codec::Decode;
+use codec::Encode;
+use sc_consensus::BlockCheckParams;
+use sc_consensus::BlockImport;
+use sc_consensus::BlockImportParams;
+use sc_consensus::ImportResult;
+use sc_consensus::Verifier;
 use sc_network::config::MultiaddrWithPeerId;
 use sc_network::config::NetworkConfiguration;
 use sc_network::config::NonReservedPeerMode;
 use sc_network::config::SyncMode;
 use sc_network::config::TransportConfig;
 use sc_network::Multiaddr;
 use sc_network::PeerId;
+use sp_consensus::error::Error as ConsensusError;
+use sp_core::RuntimeDebug;
+use sp_runtime::traits::BlakeTwo256;
+use sp_runtime::traits::Block as BlockT;
+use sp_runtime::traits::Extrinsic as ExtrinsicT;
+use sp_runtime::traits::Header as HeaderT;
+use sp_runtime::traits::Verify;
+use tracing::info;
+
+#[derive(Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, parity_util_mem::MallocSizeOf)]
+pub enum Extrinsic {
+    IncludeData(Vec<u8>),
+    StorageChange(Vec<u8>, Option<Vec<u8>>),
+}
+
+impl serde::Serialize for Extrinsic {
```

Comment:
I don't see any serialization happening in this module.
Seems this is implemented only because some trait requires it?
If so, please add a comment documenting why it exists.

Reply:
yes, I will comment it